pkg/kubectl/rolling_updater.go

```diff
@@ -208,9 +208,15 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
-		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
+	// If the desired new scale is 0, then the max unavailable is necessarily
+	// the effective scale of the old RC regardless of the configuration
+	// (equivalent to 100% maxUnavailable).
+	if desired == 0 {
+		maxUnavailable = int(math.Max(float64(oldRc.Spec.Replicas), float64(oldRc.Status.Replicas)))
```

Comment:
My memory's fuzzy but I think we compute relative to `original` as a compromise around asymmetry: with original=0, desired=10, maxUnavailable=50% we'd never scale down due to a minAvailable requirement of 5.

I added another commit which just sets maxUnavailable and minAvailable directly when desired=0 to remove all ambiguity. I'm inclined to not touch the minAvailable calculation otherwise at the moment since all the edge cases aren't in my head.